plenum/test/primary_selection/test_promotion_before_view_change.py

```diff
@@ -0,0 +1,52 @@
+from plenum.test.node_catchup.helper import ensure_all_nodes_have_same_data
+
+from plenum.test.node_catchup.test_config_ledger import start_stopped_node
+
+from plenum.test.helper import sdk_send_random_and_check, checkViewNoForNodes
+from plenum.test.pool_transactions.helper import demote_node, disconnect_node_and_ensure_disconnected, promote_node
+from plenum.test.test_node import ensureElectionsDone, checkNodesConnected
+
+nodeCount = 7
+
+
+def test_promotion_before_view_change(looper,
+                                      txnPoolNodeSet,
+                                      tdir,
+                                      tconf,
+                                      allPluginsPath,
+                                      sdk_wallet_stewards,
+                                      sdk_pool_handle):
+
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_stewards[0], 1)
+    assert txnPoolNodeSet[0].master_replica.isPrimary
+    assert txnPoolNodeSet[1].replicas._replicas[1].isPrimary
+    assert txnPoolNodeSet[2].replicas._replicas[2].isPrimary
+    starting_view_number = checkViewNoForNodes(txnPoolNodeSet)
+
+    node_2 = txnPoolNodeSet[1]
+    node_3 = txnPoolNodeSet[2]
+    node_4 = txnPoolNodeSet[3]
+
+    # Demote node 3
+    steward_3 = sdk_wallet_stewards[1]
+    demote_node(looper, steward_3, sdk_pool_handle, node_2)
+    disconnect_node_and_ensure_disconnected(looper, txnPoolNodeSet, node_2)
+    looper.removeProdable(node_2)
+    txnPoolNodeSet.remove(node_2)
+
+    # Checking that view change happened
+    ensureElectionsDone(looper, txnPoolNodeSet, instances_list=[0, 1])
```

Comment:
Should call `waitForViewChange` before `ensureElectionsDone`? `ensureElectionsDone` can be True even if no view change happened.